# GitLab NGINX Patch: Use GitLab labels and annotations for consistent Pod metadata.
diff --git a/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-createSecret.yaml b/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-createSecret.yaml
index 463ab80d7..d60bd7067 100644
--- a/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-createSecret.yaml
+++ b/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-createSecret.yaml
@@ -11,6 +11,8 @@ metadata:
     {{- toYaml . | nindent 4 }}
     {{- end }}
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: admission-webhook
     {{- with .Values.controller.admissionWebhooks.patch.labels }}
diff --git a/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-patchWebhook.yaml b/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-patchWebhook.yaml
index 862c83c1e..ab4ab64a7 100644
--- a/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-patchWebhook.yaml
+++ b/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-patchWebhook.yaml
@@ -11,6 +11,8 @@ metadata:
     {{- toYaml . | nindent 4 }}
     {{- end }}
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: admission-webhook
     {{- with .Values.controller.admissionWebhooks.patch.labels }}
diff --git a/charts/nginx-ingress/templates/clusterrolebinding.yaml b/charts/nginx-ingress/templates/clusterrolebinding.yaml
index 8f91aac80..00bc22f09 100644
--- a/charts/nginx-ingress/templates/clusterrolebinding.yaml
+++ b/charts/nginx-ingress/templates/clusterrolebinding.yaml
@@ -3,6 +3,8 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     {{- with .Values.controller.labels }}
     {{- toYaml . | nindent 4 }}
diff --git a/charts/nginx-ingress/templates/controller-configmap-addheaders.yaml b/charts/nginx-ingress/templates/controller-configmap-addheaders.yaml
index 4e4bd1310..14adb527d 100644
--- a/charts/nginx-ingress/templates/controller-configmap-addheaders.yaml
+++ b/charts/nginx-ingress/templates/controller-configmap-addheaders.yaml
@@ -3,6 +3,8 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-configmap-proxyheaders.yaml b/charts/nginx-ingress/templates/controller-configmap-proxyheaders.yaml
index 0a22600db..b49bb0c25 100644
--- a/charts/nginx-ingress/templates/controller-configmap-proxyheaders.yaml
+++ b/charts/nginx-ingress/templates/controller-configmap-proxyheaders.yaml
@@ -3,6 +3,8 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-configmap-tcp.yaml b/charts/nginx-ingress/templates/controller-configmap-tcp.yaml
index 34a2bce13..d1b99866d 100644
--- a/charts/nginx-ingress/templates/controller-configmap-tcp.yaml
+++ b/charts/nginx-ingress/templates/controller-configmap-tcp.yaml
@@ -3,6 +3,8 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-configmap-udp.yaml b/charts/nginx-ingress/templates/controller-configmap-udp.yaml
index 7137da9ad..713667553 100644
--- a/charts/nginx-ingress/templates/controller-configmap-udp.yaml
+++ b/charts/nginx-ingress/templates/controller-configmap-udp.yaml
@@ -3,6 +3,8 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-configmap.yaml b/charts/nginx-ingress/templates/controller-configmap.yaml
index b73cdc2d0..c82dc2034 100644
--- a/charts/nginx-ingress/templates/controller-configmap.yaml
+++ b/charts/nginx-ingress/templates/controller-configmap.yaml
@@ -2,6 +2,8 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-daemonset.yaml b/charts/nginx-ingress/templates/controller-daemonset.yaml
index 30c742abc..3e77b9068 100644
--- a/charts/nginx-ingress/templates/controller-daemonset.yaml
+++ b/charts/nginx-ingress/templates/controller-daemonset.yaml
@@ -3,6 +3,8 @@ apiVersion: apps/v1
 kind: DaemonSet
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-deployment.yaml b/charts/nginx-ingress/templates/controller-deployment.yaml
index deeca7cbd..d56f34904 100644
--- a/charts/nginx-ingress/templates/controller-deployment.yaml
+++ b/charts/nginx-ingress/templates/controller-deployment.yaml
@@ -3,6 +3,8 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
@@ -10,8 +12,12 @@ metadata:
     {{- end }}
   name: {{ include "ingress-nginx.controller.fullname" . }}
   namespace: {{ include "ingress-nginx.namespace" . }}
+  annotations:
+  {{- if include "gitlab.deploymentAnnotations" . }}
+    {{- include "gitlab.deploymentAnnotations" . | nindent 4 }}
+  {{- end }}
   {{- if .Values.controller.annotations }}
-  annotations: {{ toYaml .Values.controller.annotations | nindent 4 }}
+    {{ toYaml .Values.controller.annotations | nindent 4 }}
   {{- end }}
 spec:
   selector:
@@ -43,6 +49,7 @@ spec:
         {{- with .Values.controller.labels }}
         {{- toYaml . | nindent 8 }}
         {{- end }}
+      {{- include "gitlab.podLabels" . | nindent 8 }}
       {{- if .Values.controller.podLabels }}
         {{- toYaml .Values.controller.podLabels | nindent 8 }}
       {{- end }}
diff --git a/charts/nginx-ingress/templates/controller-hpa.yaml b/charts/nginx-ingress/templates/controller-hpa.yaml
index b5f81b8e7..f1380f726 100644
--- a/charts/nginx-ingress/templates/controller-hpa.yaml
+++ b/charts/nginx-ingress/templates/controller-hpa.yaml
@@ -7,6 +7,8 @@ metadata:
   annotations: {{ toYaml . | nindent 4 }}
   {{- end }}
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-ingressclass-aliases.yaml b/charts/nginx-ingress/templates/controller-ingressclass-aliases.yaml
index ffe22310c..ac8f2a67f 100644
--- a/charts/nginx-ingress/templates/controller-ingressclass-aliases.yaml
+++ b/charts/nginx-ingress/templates/controller-ingressclass-aliases.yaml
@@ -5,6 +5,8 @@ apiVersion: networking.k8s.io/v1
 kind: IngressClass
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" $ | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with $.Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-ingressclass.yaml b/charts/nginx-ingress/templates/controller-ingressclass.yaml
index 4f90d8711..971c30214 100644
--- a/charts/nginx-ingress/templates/controller-ingressclass.yaml
+++ b/charts/nginx-ingress/templates/controller-ingressclass.yaml
@@ -3,6 +3,8 @@ apiVersion: networking.k8s.io/v1
 kind: IngressClass
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-keda.yaml b/charts/nginx-ingress/templates/controller-keda.yaml
index 24d30fa0a..0a319e5c0 100644
--- a/charts/nginx-ingress/templates/controller-keda.yaml
+++ b/charts/nginx-ingress/templates/controller-keda.yaml
@@ -3,6 +3,8 @@ apiVersion: {{ .Values.controller.keda.apiVersion }}
 kind: ScaledObject
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-networkpolicy.yaml b/charts/nginx-ingress/templates/controller-networkpolicy.yaml
index e68f9916d..1ac8b6e5e 100644
--- a/charts/nginx-ingress/templates/controller-networkpolicy.yaml
+++ b/charts/nginx-ingress/templates/controller-networkpolicy.yaml
@@ -3,6 +3,8 @@ apiVersion: networking.k8s.io/v1
 kind: NetworkPolicy
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-poddisruptionbudget.yaml b/charts/nginx-ingress/templates/controller-poddisruptionbudget.yaml
index 90508d681..c7d5b2cfb 100644
--- a/charts/nginx-ingress/templates/controller-poddisruptionbudget.yaml
+++ b/charts/nginx-ingress/templates/controller-poddisruptionbudget.yaml
@@ -13,6 +13,8 @@ apiVersion: {{ template "gitlab.pdb.apiVersion" $pdbCfg }}
 kind: PodDisruptionBudget
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-prometheusrule.yaml b/charts/nginx-ingress/templates/controller-prometheusrule.yaml
index 4a9357f71..d8551a37a 100644
--- a/charts/nginx-ingress/templates/controller-prometheusrule.yaml
+++ b/charts/nginx-ingress/templates/controller-prometheusrule.yaml
@@ -9,6 +9,8 @@ metadata:
   namespace: {{ include "ingress-nginx.namespace" . }}
 {{- end }}
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
   {{- if .Values.controller.metrics.prometheusRule.additionalLabels }}
diff --git a/charts/nginx-ingress/templates/controller-role.yaml b/charts/nginx-ingress/templates/controller-role.yaml
index 127b368c4..a8522e912 100644
--- a/charts/nginx-ingress/templates/controller-role.yaml
+++ b/charts/nginx-ingress/templates/controller-role.yaml
@@ -3,6 +3,8 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: Role
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-rolebinding.yaml b/charts/nginx-ingress/templates/controller-rolebinding.yaml
index 153430aa2..7c692ee50 100644
--- a/charts/nginx-ingress/templates/controller-rolebinding.yaml
+++ b/charts/nginx-ingress/templates/controller-rolebinding.yaml
@@ -3,6 +3,8 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-secret.yaml b/charts/nginx-ingress/templates/controller-secret.yaml
index f20f53469..b891f8461 100644
--- a/charts/nginx-ingress/templates/controller-secret.yaml
+++ b/charts/nginx-ingress/templates/controller-secret.yaml
@@ -3,6 +3,8 @@ apiVersion: v1
 kind: Secret
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-service-internal.yaml b/charts/nginx-ingress/templates/controller-service-internal.yaml
index 6d0b47caf..e49c2e0db 100644
--- a/charts/nginx-ingress/templates/controller-service-internal.yaml
+++ b/charts/nginx-ingress/templates/controller-service-internal.yaml
@@ -7,6 +7,8 @@ metadata:
     {{ $key }}: {{ tpl ($value | toString) $ | quote }}
   {{- end }}
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
   {{- if .Values.controller.service.labels }}
diff --git a/charts/nginx-ingress/templates/controller-service-metrics.yaml b/charts/nginx-ingress/templates/controller-service-metrics.yaml
index 4b25a840e..e5eeb750d 100644
--- a/charts/nginx-ingress/templates/controller-service-metrics.yaml
+++ b/charts/nginx-ingress/templates/controller-service-metrics.yaml
@@ -6,6 +6,8 @@ metadata:
   annotations: {{ toYaml .Values.controller.metrics.service.annotations | nindent 4 }}
 {{- end }}
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
   {{- if .Values.controller.metrics.service.labels }}
diff --git a/charts/nginx-ingress/templates/controller-service-webhook.yaml b/charts/nginx-ingress/templates/controller-service-webhook.yaml
index 67aac0d9a..da1384b20 100644
--- a/charts/nginx-ingress/templates/controller-service-webhook.yaml
+++ b/charts/nginx-ingress/templates/controller-service-webhook.yaml
@@ -6,6 +6,8 @@ metadata:
   annotations: {{ toYaml .Values.controller.admissionWebhooks.service.annotations | nindent 4 }}
 {{- end }}
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-service.yaml b/charts/nginx-ingress/templates/controller-service.yaml
index e082c958d..f25955b8c 100644
--- a/charts/nginx-ingress/templates/controller-service.yaml
+++ b/charts/nginx-ingress/templates/controller-service.yaml
@@ -7,6 +7,8 @@ metadata:
     {{ $key }}: {{ tpl ($value | toString) $ | quote }}
   {{- end }}
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
   {{- if .Values.controller.service.labels }}
diff --git a/charts/nginx-ingress/templates/controller-serviceaccount.yaml b/charts/nginx-ingress/templates/controller-serviceaccount.yaml
index a1ebd9ea2..ae11a3992 100644
--- a/charts/nginx-ingress/templates/controller-serviceaccount.yaml
+++ b/charts/nginx-ingress/templates/controller-serviceaccount.yaml
@@ -3,6 +3,8 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- with .Values.controller.labels }}
diff --git a/charts/nginx-ingress/templates/controller-servicemonitor.yaml b/charts/nginx-ingress/templates/controller-servicemonitor.yaml
index 93ab4d242..d7f967713 100644
--- a/charts/nginx-ingress/templates/controller-servicemonitor.yaml
+++ b/charts/nginx-ingress/templates/controller-servicemonitor.yaml
@@ -9,6 +9,8 @@ metadata:
   namespace: {{ include "ingress-nginx.namespace" . }}
   {{- end }}
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
     {{- if .Values.controller.metrics.serviceMonitor.additionalLabels }}
diff --git a/charts/nginx-ingress/templates/default-backend-deployment.yaml b/charts/nginx-ingress/templates/default-backend-deployment.yaml
index e4e322d5f..a403b57ca 100644
--- a/charts/nginx-ingress/templates/default-backend-deployment.yaml
+++ b/charts/nginx-ingress/templates/default-backend-deployment.yaml
@@ -3,6 +3,8 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: default-backend
     {{- with .Values.defaultBackend.labels }}
@@ -10,6 +12,10 @@ metadata:
     {{- end }}
   name: {{ include "ingress-nginx.defaultBackend.fullname" . }}
   namespace: {{ include "ingress-nginx.namespace" . }}
+  annotations:
+  {{- if include "gitlab.deploymentAnnotations" . }}
+    {{- include "gitlab.deploymentAnnotations" . | nindent 4 }}
+  {{- end }}
 spec:
   selector:
     matchLabels:
diff --git a/charts/nginx-ingress/templates/default-backend-extra-configmaps.yaml b/charts/nginx-ingress/templates/default-backend-extra-configmaps.yaml
index 9af56cf38..5c9c39beb 100644
--- a/charts/nginx-ingress/templates/default-backend-extra-configmaps.yaml
+++ b/charts/nginx-ingress/templates/default-backend-extra-configmaps.yaml
@@ -5,6 +5,8 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" $ | nindent 4 }}
     app.kubernetes.io/component: default-backend
     {{- with $.Values.defaultBackend.labels }}
diff --git a/charts/nginx-ingress/templates/default-backend-hpa.yaml b/charts/nginx-ingress/templates/default-backend-hpa.yaml
index 03e562cb5..48375f08c 100644
--- a/charts/nginx-ingress/templates/default-backend-hpa.yaml
+++ b/charts/nginx-ingress/templates/default-backend-hpa.yaml
@@ -7,6 +7,8 @@ metadata:
   annotations: {{ toYaml . | nindent 4 }}
   {{- end }}
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: default-backend
     {{- with .Values.defaultBackend.labels }}
diff --git a/charts/nginx-ingress/templates/default-backend-networkpolicy.yaml b/charts/nginx-ingress/templates/default-backend-networkpolicy.yaml
index 90b3c2ba0..bb2e5440e 100644
--- a/charts/nginx-ingress/templates/default-backend-networkpolicy.yaml
+++ b/charts/nginx-ingress/templates/default-backend-networkpolicy.yaml
@@ -3,6 +3,8 @@ apiVersion: networking.k8s.io/v1
 kind: NetworkPolicy
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: default-backend
     {{- with .Values.defaultBackend.labels }}
diff --git a/charts/nginx-ingress/templates/default-backend-poddisruptionbudget.yaml b/charts/nginx-ingress/templates/default-backend-poddisruptionbudget.yaml
index 6b876e764..67082d634 100644
--- a/charts/nginx-ingress/templates/default-backend-poddisruptionbudget.yaml
+++ b/charts/nginx-ingress/templates/default-backend-poddisruptionbudget.yaml
@@ -9,6 +9,8 @@ apiVersion: {{ template "gitlab.pdb.apiVersion" $pdbCfg }}
 kind: PodDisruptionBudget
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: default-backend
     {{- with .Values.defaultBackend.labels }}
diff --git a/charts/nginx-ingress/templates/default-backend-service.yaml b/charts/nginx-ingress/templates/default-backend-service.yaml
index 65b6b8362..3be36c95a 100644
--- a/charts/nginx-ingress/templates/default-backend-service.yaml
+++ b/charts/nginx-ingress/templates/default-backend-service.yaml
@@ -6,6 +6,8 @@ metadata:
   annotations: {{ toYaml .Values.defaultBackend.service.annotations | nindent 4 }}
 {{- end }}
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: default-backend
     {{- with .Values.defaultBackend.labels }}
diff --git a/charts/nginx-ingress/templates/default-backend-serviceaccount.yaml b/charts/nginx-ingress/templates/default-backend-serviceaccount.yaml
index 15b1388f0..f023c054e 100644
--- a/charts/nginx-ingress/templates/default-backend-serviceaccount.yaml
+++ b/charts/nginx-ingress/templates/default-backend-serviceaccount.yaml
@@ -3,6 +3,8 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: default-backend
     {{- with .Values.defaultBackend.labels }}
diff --git a/charts/nginx-ingress/templates/default-backend-deployment.yaml b/charts/nginx-ingress/templates/default-backend-deployment.yaml
index ea84caec9..8bfc3db50 100644
--- a/charts/nginx-ingress/templates/default-backend-deployment.yaml
+++ b/charts/nginx-ingress/templates/default-backend-deployment.yaml
@@ -43,6 +43,7 @@ spec:
         {{- with .Values.defaultBackend.labels }}
         {{- toYaml . | nindent 8 }}
         {{- end }}
+      {{- include "gitlab.podLabels" . | nindent 8 }}
       {{- if .Values.defaultBackend.podLabels }}
         {{- toYaml .Values.defaultBackend.podLabels | nindent 8 }}
       {{- end }}
diff --git a/charts/nginx-ingress/templates/clusterrole.yaml b/charts/nginx-ingress/templates/clusterrole.yaml
index 51bc5002c..235eaf102 100644
--- a/charts/nginx-ingress/templates/clusterrole.yaml
+++ b/charts/nginx-ingress/templates/clusterrole.yaml
@@ -9,6 +9,8 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
   labels:
+    {{- include "gitlab.standardLabels" . | nindent 4 }}
+    {{- include "gitlab.commonLabels" . | nindent 4 }}
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     {{- with .Values.controller.labels }}
     {{- toYaml . | nindent 4 }}
diff --git a/charts/nginx-ingress/values.yaml b/charts/nginx-ingress/values.yaml
index 15aaf2ca1..efe20d2ac 100644
--- a/charts/nginx-ingress/values.yaml
+++ b/charts/nginx-ingress/values.yaml
@@ -1210,3 +1210,6 @@ dhParam: ""
 tcpExternalConfig: ""
 
 externalIpTpl: '{{ .Values.global.hosts.externalIP }}'
+
+common:
+  labels: {}
